### PR TITLE
linuxptp: shrink binaries and fix musl compile error

### PR DESCRIPTION
**Pull request by sdwalker**
Maintainer: @wodu
Compile tested: ar71xx, wndr3700v2, LEDE r865
Run tested: ar71xx, wndr3700v2, LEDE r865

Description: Binaries are significantly bloated as the -Os CFLAG is not being honoured.

### CODE REVIEW

File: net/linuxptp/patches/002-honour-COPTS.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/makefile
++++ b/makefile
+@@ -20,7 +20,7 @@ KBUILD_OUTPUT =
+ DEBUG	=
+ CC	= $(CROSS_COMPILE)gcc
+ VER     = -DVER=$(version)
+-CFLAGS	= -Wall $(VER) $(incdefs) $(DEBUG) $(EXTRA_CFLAGS)
++CFLAGS	+= -Wall $(VER) $(incdefs) $(DEBUG) $(EXTRA_CFLAGS)
```
Comment (yousong) [inline net/linuxptp/patches/002-honour-COPTS.patch]: We can avoid carrying an extra patch by `MAKE_VARS += EXTRA_CFLAGS="$(TARGET_CFLAGS) $(TARGET_CPPFLAGS)"`

**Comment by champtar** [comment]
@sdwalker see @yousong comment

**Comment by wodu** [comment]
I have just tested with MAKE_VARS in the Makefile and it seems to just do the work. Otherwise it's ok.

Even with -Os the binaries seem to take more than 270k. I will have a look at it.

**Comment by sdwalker** [comment]
@wodu I have further changes that split the monolithic package into daemon, hardware, management and synchronization packages.

**Comment by champtar** [comment]
should we merge or wait for the other changes? @sdwalker @wodu

**Comment by wodu** [comment]
For me, we could merge.

**Pull request merged**